Instructor: send submission reminders: check if session is open before sending #1883

HirdayGupta opened the pull request:
Fixes #1883

**Outline of solution:**

- Added a check in `InstructorFeedbackRemindAction` to verify that session is open and set appropriate status message if it isn't.
- Added a similar check in `InstructorFeedbackRemindParticularStudentsAction`

The status message looks like this: _[Screenshot]_ (Had to remove the disabled attributes from the remind button to get the status message to show)

<img width="1165" alt="screen shot 2017-08-18 at 9 51 36 pm" src="https://user-images.githubusercontent.com/18088721/29467932-e8bf86ea-845f-11e7-8120-3d4c5fa4e984.png">

HirdayGupta commented:
Ready for review.

HirdayGupta commented:
@damithc Sir, can label `s.ToReview`?

HirdayGupta commented:
@damithc Sir PR needs review.

damithc commented:
@HirdayGupta Is this already disabled in the front end and this is just a 2nd check at the backend?

HirdayGupta commented:
@damithc Yes sir, I believe that's how it is now. Did the issue only because of[ your comment in the issue discussion](https://github.com/TEAMMATES/teammates/issues/1883#issuecomment-68048315) that said it is better to check in the backend too.

damithc commented:
> @damithc Yes sir, I believe that's how it is now. Did the issue only because of your comment in the issue discussion that said it is better to check in the backend too.

Roger that. Waiting for next phase to finish (i.e. review by a @TEAMMATES/seniordevs)

HirdayGupta commented:
@leeyimin Please review. This PR should not take too much of your time. It's just an additional backend check for a feature that's already disabled in the UI.

leeyimin commented on src/main/java/teammates/ui/controller/InstructorFeedbackRemindAction.java:
```diff
@@ -16,11 +17,19 @@ protected ActionResult execute() {
+        if (!feedbackSession.isOpened()) {
+            statusToUser.add(new StatusMessage(
+                    Const.StatusMessages.FEEDBACK_SESSION_REMINDERSSESSIONNOTOPEN, StatusMessageColor.DANGER));
+            return createRedirectResult(nextUrl);
```
Add statusToAdmin.

HirdayGupta replied:
@leeyimin What should be the status to admin here? Would "Reminder email not sent out to the selected users as the feedback session is not open." be okay?

HirdayGupta replied:
Or would the same status as displayed to users also be fine? (The one I have added in `Const.StatusMessages.FEEDBACK_SESSION_REMINDERSSESSIONNOTOPEN`)

HirdayGupta replied:
@leeyimin If you could let me know tonight, I can have the PR ready for full review by tomorrow morning. Otherwise, I'll commit whatever I see fit, and we can take it from there. :)

leeyimin commented on src/main/java/teammates/ui/controller/InstructorFeedbackRemindParticularStudentsAction.java:
```diff
@@ -16,11 +17,19 @@ protected ActionResult execute() {
 
+        if (!feedbackSession.isOpened()) {
+            statusToUser.add(new StatusMessage(
+                    Const.StatusMessages.FEEDBACK_SESSION_REMINDERSSESSIONNOTOPEN, StatusMessageColor.DANGER));
```
Same here.

HirdayGupta commented:
@leeyimin Changes made. Updated the relevant action tests and added `statusToAdmin`. I have currently added the following statusToAdmin message: `Reminder email could not be sent out as the feedback session is not open for submissions.` for both actions. Let me know if you have any suggestions about this message. Perhaps even @damithc can weigh in here. 

Other than the above clarification regarding contents of the status message, this PR is **ready for review!**

HirdayGupta commented:
@damithc Any suggestion about the status message to admin to be shown if the session is not open and the instructor tries to send a reminder email? As of now I have set it as `Reminder email could not be sent out as the feedback session is not open for submissions.` for both `InstructorFeedbackRemind `and `InstructorFeedbackRemindParticularStudents` actions.

damithc commented:
> @damithc Any suggestion about the status message to admin to be shown if the session is not open and the instructor tries to send a reminder email? As of now I have set it as Reminder email could not be sent out as the feedback session is not open for submissions. for both InstructorFeedbackRemindand InstructorFeedbackRemindParticularStudents actions.

I'm find with your current choice @HirdayGupta

HirdayGupta commented:
Thanks @damithc sir. @leeyimin Please review.
